### ISSUE TITLE
[Tests bug] Tests exposes incorrect behavior of warmup scheduling

### ISSUE THREAD

**Issue by DrMatters** [facebookresearch/ParlAI#4243]
Fixing the **clearly** wrong parameter name in testbench (#4242) exposed that warmup schedulers are behaving incorrectly.

It seems to me that the `def step` method [here](https://github.com/facebookresearch/ParlAI/blob/89e8c323090a9ae48552b78518ea3b553474722f/parlai/nn/lr_scheduler.py#L310) of `ParlAILRScheduler` stops to warm up the learning rate too early.

I think it might be easier and less error-prone to use [`SequentialLR`](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.SequentialLR.html#torch.optim.lr_scheduler.SequentialLR) scheduler instead of custom `ParlAILRScheduler`.
